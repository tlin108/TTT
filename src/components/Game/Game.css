.status {
  margin-bottom: 10px;
}

.game {
  display: inline-flex;
  flex-direction: row;
  padding: 20px;
}

.game-info {
  margin-left: 20px;
}