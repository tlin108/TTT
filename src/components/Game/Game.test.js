import React from 'react';
import { shallow } from 'enzyme';
import Game from './index';

const wrapper = shallow(<Game />);

it('renders without crashing', () => {
  expect(wrapper).toMatchSnapshot();
});

describe('Check Winner', () => {
  const horizontalWin = 
    ['O', 'X', 'X',
     'X', null, 'X',
     'O', 'O', 'O'
    ]

  const verticalWin = 
    ['O', 'X', 'X',
     'X', null, 'X',
     'O', 'O', 'X'
    ]

  const diagonalWin = 
    ['O', null, 'X',
     null, 'X', 'X',
     'X', 'O', 'O'
    ]

  it('checks for horizontal win', () => {
    expect(wrapper.instance().calculateWinner(horizontalWin)).toEqual('O');
  })

  it('checks for vertical win', () => {
    expect(wrapper.instance().calculateWinner(verticalWin)).toEqual('X');
  })

  it('checks for diagonal win', () => {
    expect(wrapper.instance().calculateWinner(diagonalWin)).toEqual('X');
  })
})