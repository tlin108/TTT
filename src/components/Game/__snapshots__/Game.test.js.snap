exports[`test renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="game">
    <div>
        <Board
            onClick={[Function]}
            squares={
                Array [
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                    null,
                  ]
            } />
    </div>
    <div
        className="game-info">
        <div>
            Next player: X
        </div>
        <ol>
            <li>
                <a
                    href="#"
                    onClick={[Function]}>
                    Game start
                </a>
            </li>
        </ol>
    </div>
</div>,
  "nodes": Array [
    <div
      className="game">
      <div>
            <Board
                  onClick={[Function]}
                  squares={
                        Array [
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                            ]
                  } />
      </div>
      <div
            className="game-info">
            <div>
                  Next player: X
            </div>
            <ol>
                  <li>
                        <a
                              href="#"
                              onClick={[Function]}>
                              Game start
                        </a>
                  </li>
            </ol>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Game />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Game {
        "_reactInternalInstance": [Circular],
        "calculateWinner": [Function],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "history": Array [
            Object {
              "squares": Array [
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
              ],
            },
          ],
          "stepNumber": 0,
          "xIsNext": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="game">
          <div>
                    <Board
                              onClick={[Function]}
                              squares={
                                        Array [
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                ]
                              } />
          </div>
          <div
                    className="game-info">
                    <div>
                              Next player: X
                    </div>
                    <ol>
                              <li>
                                        <a
                                                  href="#"
                                                  onClick={[Function]}>
                                                  Game start
                                        </a>
                              </li>
                    </ol>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="game">
          <div>
                    <Board
                              onClick={[Function]}
                              squares={
                                        Array [
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                  null,
                                                ]
                              } />
          </div>
          <div
                    className="game-info">
                    <div>
                              Next player: X
                    </div>
                    <ol>
                              <li>
                                        <a
                                                  href="#"
                                                  onClick={[Function]}>
                                                  Game start
                                        </a>
                              </li>
                    </ol>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Game />,
}
`;
